pexec/managed_process.go

```diff
@@ -308,18 +341,32 @@ func (p *managedProcess) Stop() error {
 	// p.cmd can no longer be modified rendering it safe to read
 	// without a lock held.
 
-	p.logger.Info("stopping")
-	// First let's try to interrupt the process.
-	if err := p.cmd.Process.Signal(os.Interrupt); err != nil && !errors.Is(err, os.ErrProcessDone) {
+	p.logger.Infof("stopping process %d with %s", p.cmd.Process.Pid, p.stopSig.String())
+	// First let's try to directly signal the process.
+	if err := p.cmd.Process.Signal(p.stopSig); err != nil && !errors.Is(err, os.ErrProcessDone) {
 		return errors.Wrap(err, "error interrupting process")
 	}
 
-	// If after a while the process still isn't stopping, let's kill it.
-	timer := time.NewTimer(waitInterrupt)
+	// In case the process didn't stop, or left behind any orphan children in its process group,
+	// we now send a signal to everything in the process group after a brief wait.
+	timer := time.NewTimer(p.stopWaitInterval)
 	defer timer.Stop()
 	select {
 	case <-timer.C:
-		if err := p.cmd.Process.Kill(); err != nil && !errors.Is(err, os.ErrProcessDone) {
+		p.logger.Infof("stopping entire process group %d with %s", p.cmd.Process.Pid, p.stopSig.String())
+		if err := syscall.Kill(-p.cmd.Process.Pid, p.stopSig.(syscall.Signal)); err != nil && !errors.Is(err, os.ErrProcessDone) {
+			return errors.Wrap(err, "error interrupting process")
```

Comment:
if your log has group, pid, sig, and all that, make sure to include in both errors here in case OS does not